src/context/filterContext/filterContext.jsx

```diff
@@ -0,0 +1,34 @@
+import { useContext } from "react";
+import { createContext } from "react";
+import { useReducer } from "react";
+import { useProduct } from "../productContext/productContext";
+import { filterReducer } from '../../reducerFunctions/filterReducer/filterReducer.jsx';
+import { sortDataByRange } from '../../reducerFunctions/filterReducer/reducerUtilities/filterSortFunc';
+import { filterByRating } from '../../reducerFunctions/filterReducer/reducerUtilities/filterRatingFunc';
+import { filterByPrice } from '../../reducerFunctions/filterReducer/reducerUtilities/filterPriceFunc';
+import { filterDataByCategory } from '../../reducerFunctions/filterReducer/reducerUtilities/filterCategoryFunc';
+import { useEffect } from "react";
+
+
+
+const FilterContext = createContext(null);
+
+const FilterProvider = ({ children }) => {
+    const { productList } = useProduct();
+    const [state, dispatch] = useReducer(filterReducer, { sortbyrange: "", sortByRating: "", priceValue: 160000, category: { isBike: false, isScooter: false } });
```

Comment:
You can extract out the initialState and then you can use it for clear filter also.